src/api/types/types.ts

```diff
@@ -19,3 +22,25 @@ export type DataForQuery<TQueryKey extends keyof AxiosQueriesType> = Unwrap<
 >;
 
 export type ArgsForQuery<TQueryKey extends keyof AxiosQueriesType> = GetQueryParams<TQueryKey>;
+
+export type APIErrorOutput = Pick<AxiosError, 'code' | 'message' | 'status'> & { originalError: AxiosError };
+
+export type ClientErrorResponse<TErrorData = APIErrorOutput> = AxiosError<TErrorData, unknown>;
+
+export type ExtendedQueryMeta = QueryMeta & {
+  error: { excludedCodes: string[]; showGlobalError: boolean };
+};
+
+export type ExtendedAxiosRequestConfig = AxiosRequestConfig & {
+  _retry?: boolean;
+};
+
+export interface ErrorHandlingStrategy<T, D> {
+  getBaseErrorObject(error: T): T;
```

Comment:
do we need getBaseErrorObject getErrorObject those two methods, they literally can be anything at this point, won't narrowErrorData suffice?

Reply:
Refactored as we discussed on slack.